spock/plugins/helpers/physics.py

```diff
@@ -79,87 +77,91 @@ class PhysicsPlugin(PluginBase):
-            self.vec.y = 0
-        if self.block_collision(cb, y=-1):  # we check below feet
-            self.pos.on_ground = True
-            self.vec.y = 0
-            self.pos.y = cb.y
-        else:
-            self.pos.on_ground = False
-            self.vec -= Vector3(0, PLAYER_ENTITY_GAV, 0)
-            self.apply_vertical_drag()
-        # feet or head collide with x
-        if self.block_collision(cb, x=1) or \
-                self.block_collision(cb, x=-1) or \
-                self.block_collision(cb, y=1, x=1) or \
-                self.block_collision(cb, y=1, x=-1):
-            self.vec.x = 0
-            # replace with real info in event
-            self.event.emit("phy_collision", "x")
-        # feet or head collide with z
-        if self.block_collision(cb, z=1) or \
-                self.block_collision(cb, z=-1) or \
-                self.block_collision(cb, y=1, z=1) or \
-                self.block_collision(cb, y=1, z=-1):
-            self.vec.z = 0
-            # replace with real info in event
-            self.event.emit("phy_collision", "z")
-
-    def block_collision(self, cb, x=0, y=0, z=0):
-        block_id, meta = self.world.get_block(cb.x + x, cb.y + y, cb.z + z)
-        block = mapdata.get_block(block_id, meta)
-        if block is None:
-            return False
-        # possibly we want to use the centers of blocks as the starting
-        # points for bounding boxes instead of 0,0,0 this might make thinks
-        # easier when we get to more complex shapes that are in the center
-        # of a block aka fences but more complicated for the player uncenter
-        # the player position and bump it up a little down to prevent
-        # colliding in the floor
-        pos1 = Position(self.pos.x - self.playerbb.w / 2, self.pos.y - 0.2,
-                        self.pos.z - self.playerbb.d / 2)
-        bb1 = self.playerbb
-        bb2 = block.bounding_box
-        if bb2 is not None:
-            pos2 = Position(cb.x + x + bb2.x, cb.y + y + bb2.y,
-                            cb.z + z + bb2.z)
-            if ((pos1.x + bb1.w) >= (pos2.x) and (pos1.x) <= (
-                    pos2.x + bb2.w)) and (
-                (pos1.y + bb1.h) >= (pos2.y) and (pos1.y) <= (
-                    pos2.y + bb2.h)) and (
-                (pos1.z + bb1.d) >= (pos2.z) and (pos1.z) <= (
-                    pos2.z + bb2.d)):
-                return True
-        return False
-
-    def apply_vertical_drag(self):
-        self.vec.y -= self.vec.y * PLAYER_ENTITY_DRG
+        self.vec.y -= PLAYER_ENTITY_GAV
+        self.apply_drag()
+        mtv = self.get_mtv()
+        self.pos.on_ground = mtv.y > 0
+        self.apply_vector(mtv)
 
-    def apply_horizontal_drag(self):
+    def apply_drag(self):
+        self.vec.y -= self.vec.y * PLAYER_ENTITY_DRG
         self.vec.x -= self.vec.x * PLAYER_GND_DRG
         self.vec.z -= self.vec.z * PLAYER_GND_DRG
```

Comment:
No, the acceleration and drag both on and off ground is outright wrong currently. That will need to be fixed but for now this is Close Enough™ to allow for implementation of other plugins.